libavcodec/vvc/vvc_ps.c

```diff
@@ -1012,6 +1029,43 @@ static int sps_parse_vui(VVCSPS *sps, GetBitContext *gb, void *log_ctx)
     return 0;
 }
 
+static int sps_parse_extension(VVCSPS *sps, GetBitContext *gb, void *log_ctx)
+{
+    int range_extension_flag;
+    sps->extended_precision_flag = 0;
+    sps->ts_residual_coding_rice_present_in_sh_flag = 0;
+    sps->rrc_rice_extension_flag = 0;
+    sps->persistent_rice_adaptation_enabled_flag = 0;
+    sps->reverse_last_sig_coeff_enabled_flag = 0;
+
+    if (get_bits1(gb)) {  // sps_extension_present_flag
+        range_extension_flag = get_bits1(gb);
+        skip_bits(gb, 7);  // sps_extension_7bits
+        if (range_extension_flag) {
+            sps->extended_precision_flag = get_bits1(gb);
+            if (sps->transform_skip_enabled_flag) {
+                sps->ts_residual_coding_rice_present_in_sh_flag = get_bits1(gb);
+            }
+            sps->rrc_rice_extension_flag = get_bits1(gb);
+            sps->persistent_rice_adaptation_enabled_flag = get_bits1(gb);
+            sps->reverse_last_sig_coeff_enabled_flag = get_bits1(gb);
+        }
+    }
+
+    // Derived values
+    sps->log2_transform_range = sps->extended_precision_flag
+                              ? FFMAX(15, FFMIN(20, sps->bit_depth + 6))
+                              : 15;
+    sps->coeff_min = -(1 << (sps->extended_precision_flag
```

Comment:
maybe you can use  -(1 << sps->log2_transform_range)

Reply:
I thought this too, we could even use `av_clip_intp2(val, sps->log2_transform_range)` – the spec does not define CoeffMin and CoeffMax with reference to Log2TransformRange though so I thought it best to copy that exactly.